include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -5276,8 +5276,8 @@ namespace internal
   template <int dim, typename Number>
   struct EvaluatorTensorProduct<evaluate_general,dim,-1,0,Number>
   {
-    static const unsigned int dofs_per_cell = numbers::invalid_unsigned_int;
-    static const unsigned int n_q_points = numbers::invalid_unsigned_int;
+    static const unsigned int dofs_per_cell = 0;
+    static const unsigned int n_q_points = 0;
```

Comment:
The problem was that MSVC complained about the following use case:
```
if (/*unsigned int */ temp_size < 100)
  {
    ptr1 = &temp[0];
    ptr2 = ptr1 + temp_size;
  }
else
  {
    ptr1 = somewhere_else;
    ptr2 = ptr1 + (temp_size < numbers::invalid_unsigned_int ? temp_size : size_from_somewhere_else);
  }
```
I could of course try to include another check for `temp_size` into the function but I'm not sure this helps...